pkg/stub/handler.go

```diff
@@ -145,45 +147,92 @@ func (h *Handler) prepWatchEvent(kind, name string, annotations map[string]strin
 	return srcfg, filePath, true
 }
 
-func (h *Handler) processImageStreamWatchEvent(is *imagev1.ImageStream) {
+func (h *Handler) processImageStreamWatchEvent(is *imagev1.ImageStream) error {
 	h.mutex.Lock()
 	defer h.mutex.Unlock()
 
 	srcfg, filePath, ok := h.prepWatchEvent("imagestream", is.Name, is.Annotations)
 	if !ok {
-		return
+		if srcfg != nil {
+			logrus.Debugf("checking tag spec/status for %s spec len %d status len %d", is.Name, len(is.Spec.Tags), len(is.Status.Tags))
+			// won't be updating imagestream this go around, which sets pending==true, so see if we should turn off pending
+			pending := false
+			if len(is.Spec.Tags) == len(is.Status.Tags) {
+				for _, specTag := range is.Spec.Tags {
+					matched := false
+					for _, statusTag := range is.Status.Tags {
+						logrus.Debugf("checking spec tag %s against status tag %s with num items %d", specTag.Name, statusTag.Tag, len(statusTag.Items))
+						if specTag.Name == statusTag.Tag {
+							for _, event := range statusTag.Items {
+								logrus.Debugf("checking status tag %d against spec tag %#v", event.Generation, specTag.Generation)
+								if specTag.Generation != nil &&
+									*specTag.Generation <= event.Generation {
```

Comment:
were you able to confirm this is a valid way to check the import status of a tag?  it was a bit of a guess on my part, it's been a while since i've dug through the imagestreamimport code flow.

Reply:
Yeah it definitely looked valid.  There was a gradual progression of imagestreams getting to completion state as images were downloaded.  Working from home yesterday, with a slower network, made that progression even more deliberate and pronounced.  An eventually all the tags matched up.

I was able to watch specific image streams via `oc` as well as monitor the samplesresource obj via `oc` and run with debug in the samples operator pod, and line things up.  And as I mentioned in the description, while my pod debug showed the clusteroperator api calls flowing and completing without error, and the final updates setting progressing to false after it had been true when the imagestream updates started, there was no clusteroperator object to look at given the in between state that api is in based on what you and Oleg found.

Doing the `>=` check vs. `==` check was even spot on, as the `latest` tag often times was a tag ref vs. an image ref, so you would have cases where the spec generation was 1, but the status generation was 2, for example.